src/Model/UserItem.php

```diff
@@ -206,6 +216,10 @@ private static function checkShared(array $item, int $uid)
 			return true;
 		}
 
+		if ($item['gravity'] != GRAVITY_PARENT) {
```

Comment:
I can't select it directly in my review, but I believe the comment on line 214 should be changed to include shared posts and a comment should be added above this check to explain its use.